logstash_async/handler.py

```diff
@@ -132,19 +199,6 @@ def _worker_thread_is_running():
 
         return False
 
-    # ----------------------------------------------------------------------
-    def _format_record(self, record):
-        self._create_formatter_if_necessary()
-        formatted = self.formatter.format(record)
-        if isinstance(formatted, str):
-            formatted = formatted.encode(self._encoding)  # pylint: disable=redefined-variable-type
-        return formatted + b'\n'
-
-    # ----------------------------------------------------------------------
-    def _create_formatter_if_necessary(self):
-        if self.formatter is None:
-            self.formatter = LogstashFormatter()
-
     # ----------------------------------------------------------------------
     def close(self):
```

Comment:
`close()` and `_close_transport()` seem to be duplicated in both classes, I'd suggest to define both only in `SynchronousLogstashHandler`.

Reply:
Good catch. Has been implemented.